api/prometheus/v1/api.go

```diff
@@ -734,5 +753,11 @@ func (c apiClient) Do(ctx context.Context, req *http.Request) (*http.Response, [
 		}
 	}
 
+	if err == nil {
+		if len(result.Warnings) != 0 {
+			err = Warnings(result.Warnings)
+		}
+	}
+
```

Comment:
```
if err == nil {
		if len(result.Warnings) != 0 {
			err = Warnings(result.Warnings)
		}
	}
```
my idea was instead of this to just return 

`func (c apiClient) Do(ctx context.Context, req *http.Request) (*http.Response, []byte,Warnings, error){


	return resp, []byte(result.Data),Warnings, err
}